bezier-rs/lib/src/subpath/core.rs

```diff
@@ -0,0 +1,94 @@
+use super::*;
+
+/// Functionality relating to core `Subpath` operations, such as constructors and `iter`.
+impl Subpath {
+	/// Create a new `Subpath` using a list of [ManipulatorGroup]s.
+	/// A `Subpath` with less than 2 [ManipulatorGroup]s may not be closed.
+	pub fn new(manipulator_groups: Vec<ManipulatorGroup>, closed: bool) -> Subpath {
+		assert!(!closed || manipulator_groups.len() > 1);
+		Subpath { manipulator_groups, closed }
+	}
+
+	/// Create a `Subpath` consisting of 2 manipulator groups from a `Bezier`.
+	pub fn from_bezier(bezier: Bezier) -> Self {
+		Subpath::new(
+			vec![
+				ManipulatorGroup {
+					anchor: bezier.start(),
+					in_handle: None,
+					out_handle: bezier.handle_start(),
+				},
+				ManipulatorGroup {
+					anchor: bezier.end(),
+					in_handle: bezier.handle_end(),
+					out_handle: None,
+				},
+			],
+			false,
+		)
+	}
+
+	/// Returns true if and only if the `Subpath` contains at least one [ManipulatorGroup].
+	pub fn is_empty(&self) -> bool {
+		self.manipulator_groups.is_empty()
+	}
+
+	/// Returns the number of [ManipulatorGroup]s contained within the `Subpath`.
+	pub fn len(&self) -> usize {
+		self.manipulator_groups.len()
+	}
+
+	/// Returns an iterator of the [Bezier]s along the `Subpath`.
+	pub fn iter(&self) -> SubpathIter {
+		SubpathIter { sub_path: self, index: 0 }
+	}
+
+	/// Returns an SVG representation of the `Subpath`.
+	pub fn to_svg(&self, options: ToSVGOptions) -> String {
+		if self.is_empty() {
+			return String::new();
+		}
+
+		let subpath_options = format!(r#"stroke="{}" stroke-width="{}" fill="none""#, options.curve_stroke_color, options.curve_stroke_width);
```

Comment:
Sorry, I'm a bit confused by what you meant by `let svg_builder = |stroke_width: f32, fill... |`. Do you want me to just create helper functions, or an `svg_builder` class? I'm not quite familiar with what that `| ... |` notation implies. What are your thoughts on adding helper functions to `ToSVGOptions` since all the options are built into that struct anyway?

Reply:
The `| ... |` represents a closure. https://doc.rust-lang.org/book/ch13-01-closures.html

It is just a way of writing an anonymous function for simplicity. Add helper functions where you think they would be best suited, `ToSVGOptions` sounds fine to me as long as it doesn't feel overloaded.